interrupt/maven/src/main/java/iotdk/example/Interrupt.java

```diff
@@ -65,21 +65,31 @@ public void run() {
         "You may need to modify the MRAA/UPM initialization code to " +
         "ensure it works properly on your platform.\n\n";
 
+
+    public static void checkRoot(){
+      String username = System.getProperty("user.name");
+      System.out.println(username);
+      String message = "This project uses Mraa I/O operations, but you're not running as 'root'.\n"+
+      "The IO operations below might fail.\nSee the project's Readme for more info.\n\n";
+      if(!username.equals("root"))
+      {
+        System.out.println(message);
+      }
+    }
+
     public static void main(String[] args) {
+
+        checkRoot();
+
         Platform platform = mraa.getPlatformType();
         int pinNumber = 13;
 
-        switch (platform) {
-            case Platform.INTEL_MINNOWBOARD_MAX:
-            case Platform.INTEL_JOULE_EXPANSION:
-                pinNumber = 26;
-                break;
-            case Platform.UNKNOWN_PLATFORM:
-                System.err.println(unknownPlatformMessage);
-                break;
-            default:
-                break;
-        }
+        if(platform.equals(Platform.INTEL_MINNOWBOARD_MAX))
```

Comment:
These are deprecated platforms, you can remove these entirely and replace it with 'INTEL_UP2'. 
It is upto you to do it with this PR or another.

Reply:
Pin 13 will work on UP2 and most boards, even legacy ones except Minnowboard & Joule. Thus I would keep them as is for legacy support.

Reply:
IEI_TANK support will be added according https://jira01.devtools.intel.com/browse/ISSDEV-2045